lib/Sema/ConstraintSystem.cpp

```diff
@@ -2454,8 +2454,13 @@ Type ConstraintSystem::simplifyTypeImpl(Type type,
         auto *proto = assocType->getProtocol();
         auto conformance = DC->getParentModule()->lookupConformance(
           lookupBaseType, proto);
-        if (!conformance)
+        if (!conformance) {
+          if (shouldAttemptFixes() &&
+              getPhase() == ConstraintSystemPhase::Solving)
```

Comment:
Nit: I think it would be worth it to add a comment about why we do this.

Reply:
Done!

Reply:
Thank you!